From dc6ae0b8d238d061976c6671d3e867ddd43d87e1 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Tue, 2 Jan 2018 09:34:11 +0000
Subject: [PATCH] wr-policy: allow secadm_t read_policy

Port fixes from https://github.com/fedora-selinux/selinux-policy:
  2adfaf11 Add back userdom_security_admin_template() interface

Remove issue:
  $ newrole -r secadm_r -- -c "sesearch -A -s mount_t -t shell_exec_t -c file"
  [Errno 13] Permission denied: '/sys/fs/selinux/policy'

Fix avc denial:
  avc: denied { read_policy } for pid=7530 comm="python" \
  scontext=root:secadm_r:secadm_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:security_t:s15:c0.c1023 tclass=security

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/system/userdomain.if | 1 +
 1 file changed, 1 insertion(+)

diff --git a/policy/modules/system/userdomain.if b/policy/modules/system/userdomain.if
index fce30a1..5a28862 100644
--- a/policy/modules/system/userdomain.if
+++ b/policy/modules/system/userdomain.if
@@ -1445,6 +1445,7 @@ template(`userdom_security_admin_template',`
 	selinux_set_enforce_mode($1)
 	selinux_set_all_booleans($1)
 	selinux_set_parameters($1)
+	selinux_read_policy($1)
 
 	files_relabel_non_auth_files($1)
 	auth_relabel_shadow($1)
-- 
2.7.4

